odl/util/utility.py

```diff
@@ -178,18 +198,49 @@ class npy_printoptions(object):
     ...     print(np.array([np.nan, 1.00001]))
     [  whoah!  1.00001]
     """
+    orig_opts = np.get_printoptions()
 
-    def __init__(self, **extra_opts):
-        self.extra_opts = extra_opts
-        self.orig_opts = np.get_printoptions()
-
-    def __enter__(self):
-        new_opts = self.orig_opts.copy()
-        new_opts.update(self.extra_opts)
+    try:
+        new_opts = orig_opts.copy()
+        new_opts.update(extra_opts)
         np.set_printoptions(**new_opts)
+        yield
 
-    def __exit__(self, type, value, traceback):
-        np.set_printoptions(**self.orig_opts)
+    finally:
+        np.set_printoptions(**orig_opts)
+
+
+@contextmanager
+def npy_error_handling(**options):
```

Comment:
This might actually be equivalent to `np.errstate`, can be removed in that case.

Reply:
It is, function removed.